### PR TITLE
Quick Replies for VK

### PR DESCRIPTION
**Pull request by Robi9**
This PR is about supporting quick replies to VK handler. VK treats quick replies as keyboard, [here](https://vk.com/dev.php?f=4.%20Bot%20keyboards&method=bots_docs_3) we can see in more detail about keyboards.

### CODE REVIEW

File: handlers/vk/vk_test.go [hunk 1]
```diff
@@ -445,6 +472,25 @@ var sendTestCases = []ChannelSendTestCase{
 			},
 		},
 	},
+	{
+		Label:        "Send keyboard",
+		Text:         "Send keyboard",
+		URN:          "vk:123456789",
+		QuickReplies: []string{"Yes"},
```
Comment (norkans7) [inline handlers/vk/vk_test.go:479]: Can you make the test with many quick replies?

**Comment by norkans7** [comment]
This will add all  buttons in one row, I think we need to probably make a new row every 3 buttons or better use the length of the button text to know when to make a new row

**Comment by Robi9** [comment]
I did some tests and regardless of the text size, in a new line, 5 buttons are accepted. When the text on these buttons is large, they end up showing a small portion of that text and hiding the rest.

File: handlers/vk/vk.go [hunk 2]
```diff
@@ -417,6 +445,58 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
...
+		keyboard.Buttons = append(keyboard.Buttons, row)
+
+	}
+	keyboard.One_Time = true
+
+	return keyboard
+}
+
+func AddTextButton(label string, payload interface{}, color string, keyboard Keyboard) Keyboard {
+	b, err := json.Marshal(payload)
```
Comment (rowanseymour) [inline handlers/vk/vk.go:480]: I'm fine with not error checking JSON marshalling of things we control - but use `jsonx.MustMarshal`

File: handlers/vk/vk.go [hunk 3]
```diff
@@ -417,6 +445,58 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 	return status, nil
 }
 
+// ToJSON returns the JSON encoding of Keyboard.
+func ToJSON(keyboard Keyboard) string {
+	b, _ := json.Marshal(keyboard)
+	return string(b)
+}
```
Comment (rowanseymour) [inline handlers/vk/vk.go:452]: this function doesn't feel necessary - use `jsonx.MustMarshal`

File: handlers/vk/vk.go [hunk 4]
```diff
@@ -417,6 +445,58 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
...
+
+//keyboard
+func buildQuickRepliesParams(msg courier.Msg) Keyboard {
+
+	var keyboard Keyboard
+	keyboard = AddRowKeyboard(keyboard)
+
+	for _, qr := range msg.QuickReplies() {
+		keyboard = AddTextButton(qr, qr, "secondary", keyboard)
+	}
```
Comment (rowanseymour) [inline handlers/vk/vk.go:462]: maybe we can do something similar to the telegram handler and use `utils.StringsToRows` to create better layouts when things don't fit on one row?

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Comment by Robi9** [comment]
@rowanseymour I sought the same approach taken for the telegram, it applied well to the VK.

**Review by norkans7 — APPROVED** [review]

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**